src/algorithms/string/knuth-morris-pratt/knuthMorrisPratt.js

```diff
@@ -30,6 +30,10 @@ function buildPatternTable(word) {
  * @return {number}
  */
 export default function knuthMorrisPratt(text, word) {
+  if (word.length === 0) {
```

Comment:
Please take a look at the comments to the tests above. No need to do this checking.